src/GovUk.Education.ExploreEducationStatistics.Admin/Services/ImportService.cs

```diff
@@ -70,11 +66,25 @@ public class ImportService : IImportService
                 numRows,
                 message);
 
-            pQueue.AddMessage(new CloudQueueMessage(JsonConvert.SerializeObject(message)));
+            await _queueService.AddMessageAsync("imports-pending", message);
 
             _logger.LogInformation($"Sent import message for data file: {dataFileName}, releaseId: {releaseId}");
         }
 
+        public Task<Either<ActionResult, Unit>> CancelImport(ReleaseFileImportInfo import)
+        {
+            return _userService
+                .CheckCanCancelFileImport(import)
+                .OnSuccessVoid(async () =>
+                {
+                    await _queueService.AddMessageAsync("imports-cancelling", new CancelImportMessage
+                    {
+                        ReleaseId = import.ReleaseId,
+                        DataFileName = import.DataFileName
```

Comment:
I think we'd eventually like to get this working with file reference id's rather than filenames which rely on other things working to ensure they are unique. I suppose it's quite a big change at this stage to use the file id here and later lookup the filename if that's still required?

Reply:
I'd happily look at doing this, perhaps in my follow-up PR?

Reply:
That would be really cool thanks, because I see a lot of stuff referencing filename. Processor function can have a connection to the content database if it doesn't already so it can lookup files by id if it needs to and the file id on the import row would be handy!